accord-core/src/main/java/accord/coordinate/CoordinatePreAccept.java

```diff
@@ -142,7 +142,7 @@ void onNewEpochTopologyMismatch(TopologyMismatch mismatch)
         proposeInvalidate(node, node.uniqueTimestamp(Ballot::fromValues), txnId, route.homeKey(), (outcome, failure) -> {
             if (failure != null)
                 mismatch.addSuppressed(failure);
-            setFailure(mismatch);
+            callback.accept(null, mismatch);
```

Comment:
so looking at the history this logic was changed in `CASSANDRA-20707` which added a bug due to `setFailure` still existing?  This patch doesn't remove and intellij is telling me that there are still callers to `setFailure`, why would you call `setFailure` in one code path and another directly notify the callback?  For example, this no longer updates the `eventListener` and doesn't set `isDone` (if we trigger the callback are we not actually done?)

Reply:
That is to say more specifically, this is invoked by `onPreAcceptedOrNewEpoch` and is the parallel work for rejecting the query, but the `CoordinatePreAccept` phase is logically done by this point and ownership is passed to the `notAccept` / `acceptInvalidate` phase.

Reply:
> It is a callback from another process that occurs after isDone is already set.

that doesn't answer my question.  This change is taking 1 line from `setFailure` and ignoring the rest of the logic (such as monitoring).  If this change actually matters it implies that it was called *after* `isDone == true` *or* it isn't desirable to have `isDone = true` here (which feels weird as we are notifying the callback of a result).

Which is it?  If we feel its best to special case can we document the *why*?

Reply:
thanks, i just found it.  `accord.coordinate.AbstractCoordinatePreAccept#onPreAcceptedOrNewEpoch` sets `isDone` but doesn't notify the callback.